src/arkhamdb/DeckImporterMain.ttslua

```diff
@@ -399,50 +398,34 @@ function loadCards(slots, investigatorId, playerColor, commandManager, configura
     handleUnderworldMarket(cardsToSpawn, playerColor)
     handleHunchDeck(investigatorId, cardsToSpawn, playerColor)
 
-    -- Count the number of cards in each zone so we know if it's a deck or card.
-    -- TTS's Card vs. Deck distinction requires this since we can't spawn a deck with only one card
-    local zoneCounts = getZoneCounts(cardsToSpawn)
-    local zoneDecks = {}
-    for zone, count in pairs(zoneCounts) do
-      if count > 1 then
-        zoneDecks[zone] = buildDeckDataTemplate()
-      end
-    end
-    -- For each card in a deck zone, add it to that deck. Otherwise, spawn it directly
-    for _, spawnCard in ipairs(cardsToSpawn) do
-      if zoneDecks[spawnCard.zone] ~= nil then
-        addCardToDeck(zoneDecks[spawnCard.zone], spawnCard.data)
-      else
-        local cardPos = Zones.getZonePosition(playerColor, spawnCard.zone)
-        cardPos.y = 2
-        spawnObjectData({
-          data = spawnCard.data,
-          position = cardPos,
-          rotation = Zones.getDefaultCardRotation(playerColor, spawnCard.zone),
-        })
-      end
-    end
-    -- Spawn each of the decks
-    for zone, deck in pairs(zoneDecks) do
+    -- Split the card list into separate lists for each zone
+    local zoneDecks = buildZoneLists(cardsToSpawn)
+    -- Spawn the list for each zone
+    for zone, zoneCards in pairs(zoneDecks) do
       local deckPos = Zones.getZonePosition(playerColor, zone)
       deckPos.y = 3
-      local spreadCallback = nil;
+
+      local spreadCallback = nil
+      -- If cards are spread too close together TTS groups them weirdly, selecting multiples
+      -- when hovering over a single card.  This distance is the minimum to avoid that
+      local spreadDistance = 1.15
       if (zone == "SetAside4") then
         -- SetAside4 is reserved for customization cards, and we want them spread on the table
         -- so their checkboxes are visible
         if (playerColor == "White") then
-          deckPos.z = deckPos.z + (#deck.ContainedObjects - 1)
+          deckPos.z = deckPos.z + (#zoneCards - 1) * spreadDistance
         elseif (playerColor == "Green") then
-          deckPos.x = deckPos.x + (#deck.ContainedObjects - 1)
+          deckPos.x = deckPos.x + (#zoneCards - 1) * spreadDistance
         end
-        spreadCallback = function(deck) deck.spread(1.0) end
+        spreadCallback = function(deck) deck.spread(spreadDistance) end
       end
-      spawnObjectData({
-        data = deck,
-        position = deckPos,
-        rotation = Zones.getDefaultCardRotation(playerColor, zone),
-       callback_function = spreadCallback
-      })
+      Spawner.spawnCards(
+          zoneCards,
+          deckPos,
+          Zones.getDefaultCardRotation(playerColor, zone),
+          true,
+          spreadCallback)
+
```

Comment:
comment with description of arguments might be useful (so you don't need to open the required file when sth needs to change)

Reply:
Added for the sort, but I think the names for the rest of them are clear (cardList, pos, rotation, callback)